### PR TITLE
Validate state utility method

### PR DESCRIPTION
**Pull request by davidbenjamin**
@lbergelson this is the method I was talking about.

### CODE REVIEW

**Comment by davidbenjamin** [comment]
@lbergelson May I assign you?

File: src/main/java/org/broadinstitute/hellbender/tools/picard/vcf/concordance/GenotypeConcordance.java [hunk 1]
```diff
@@ -342,19 +343,15 @@ final TruthAndCallStates determineState(final VariantContext truthContext, final
         String truthAllele2 = null;
         if (null == truthState) {
             // Truth State not yet determined - will need to use truth genotypes below
-            if (truthGenotype.getAlleles().size() != 2) {
-                throw new IllegalStateException("Genotype for Variant Context: " + truthContext + " does not have exactly 2 alleles");
-            }
+            Utils.validateState(truthGenotype.getAlleles().size() == 2,"Genotype for Variant Context: " + truthContext + " does not have exactly 2 alleles");
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/tools/picard/vcf/concordance/GenotypeConcordance.java]: Shouldn't this one use the `() -> message` form?

Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/picard/vcf/concordance/GenotypeConcordance.java]: Yes, it should.

File: src/main/java/org/broadinstitute/hellbender/tools/picard/vcf/concordance/GenotypeConcordance.java [hunk 2]
```diff
@@ -342,19 +343,15 @@ final TruthAndCallStates determineState(final VariantContext truthContext, final
...
             truthAllele2 = truthGenotype.getAllele(1).getBaseString();
         }
 
         String callAllele1 = null;
         String callAllele2 = null;
         if (null == callState) {
-            if (callGenotype.getAlleles().size() != 2) {
-                throw new IllegalStateException("Genotype for Variant Context: " + callContext + " does not have exactly 2 alleles");
-            }
+            Utils.validateState(callGenotype.getAlleles().size() == 2, "Genotype for Variant Context: " + callContext + " does not have exactly 2 alleles");
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/picard/vcf/concordance/GenotypeConcordance.java]: done

File: src/main/java/org/broadinstitute/hellbender/tools/picard/vcf/concordance/GenotypeConcordance.java [hunk 3]
```diff
@@ -450,19 +447,15 @@ else if (truthRef.length() > callRef.length()) {
                 callState = CallState.getHet(allele0idx, allele1idx);
             }
 
-            if (null == callState) {
-                throw new IllegalStateException("This should never happen...  Could not classify the call variant: " + callGenotype);
-            }
+            Utils.validateState(callState != null, "This should never happen...  Could not classify the call variant: " + callGenotype);
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/picard/vcf/concordance/GenotypeConcordance.java]: done

File: src/main/java/org/broadinstitute/hellbender/utils/Utils.java [hunk 4]
```diff
@@ -683,6 +683,18 @@ public static void validateArg(final boolean condition, final Supplier<String> m
         }
     }
 
+    public static void validateState(final boolean condition, final String msg){
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/utils/Utils.java]: done

File: src/main/java/org/broadinstitute/hellbender/tools/picard/vcf/concordance/GenotypeConcordance.java (hunk 1 above)
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/picard/vcf/concordance/GenotypeConcordance.java]: done

File: src/main/java/org/broadinstitute/hellbender/tools/picard/vcf/concordance/GenotypeConcordance.java (hunk 2 above)
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/picard/vcf/concordance/GenotypeConcordance.java]: `assert` is reserved.  How about `assertion`?

**Comment by davidbenjamin** [comment]
Back to you.

**Comment by lbergelson** [comment]
@davidbenjamin Why `assertion` vs `assert`?  I think it reads better as the more active verb. This is just quibbling though.  @droazen What color should we paint the bike shed?

**Comment by davidbenjamin** [comment]
@lbergelson Because `assert` is a reserved keyword.  I would also prefer the verb.

**Comment by lbergelson** [comment]
Ah, that's a good reason not to use it.  How about just plain old `validate`?

**Comment by davidbenjamin** [comment]
@lbergelson changed it to `validate`.  Anything else?

**Pull request merged**